src/miral/basic_window_manager.cpp

```diff
@@ -1127,6 +1137,54 @@ void miral::BasicWindowManager::place_and_size(WindowInfo& root, Point const& ne
     move_tree(root, new_pos - root.window().top_left());
 }
 
+void miral::BasicWindowManager::place_attached(WindowInfo& root)
+{
+    // We can't place the attached window if there aren't any outputs
+    if (root.type() != mir_window_type_attached || outputs.size() == 0)
```

Comment:
Likewise, `root.type() != mir_window_type_attached` here is a precondition failure.